homeassistant/components/cert_expiry/helper.py

```diff
@@ -18,36 +18,38 @@
     ValidationFailure,
 )
 
+_LOGGER = logging.getLogger(__name__)
+
 
-async def async_get_cert(
-    hass: HomeAssistant,
+def async_get_cert(
     host: str,
     port: int,
-) -> dict[str, Any]:
+) -> OpenSSL.crypto.X509 | None:
     """Get the certificate for the host and port combination."""
-    async with asyncio.timeout(TIMEOUT):
-        transport, _ = await hass.loop.create_connection(
-            asyncio.Protocol,
-            host,
-            port,
-            ssl=get_default_context(),
-            happy_eyeballs_delay=0.25,
-            server_hostname=host,
-        )
+
+    context = get_default_context()
+    context.check_hostname = False
+    context.verify_mode = ssl.CERT_NONE
+    context.verify_flags = context.verify_flags | ssl.VERIFY_CRL_CHECK_CHAIN
+
+    conn = socket.create_connection((host, port), timeout=TIMEOUT)
+    sock = context.wrap_socket(conn, server_hostname=host)
```

Comment:
I might need to look at this more closely to be sure, but wouldn't you be able to do something like:
```python
    context = get_default_context()
    context.check_hostname = False
    context.verify_mode = ssl.CERT_NONE
    context.verify_flags = context.verify_flags | ssl.VERIFY_CRL_CHECK_CHAIN
    async with asyncio.timeout(TIMEOUT):
        transport, _ = await hass.loop.create_connection(
            asyncio.Protocol,
            host,
            port,
            ssl=context,
            happy_eyeballs_delay=0.25,
            server_hostname=host,
        )
```

and keep the function async?

Reply:
This doesn't seem to work as there's no way to access the unparsed certificate it seems.

Reply:
I'll get to it on the weekend or this evening/night. Probably other than that I just replaced the python files from my prod home assistant with these and test against some of my services. You just need a host with a self signed cert, plenty of guides online about that I guess.